Xamarin.ExposureNotification/ExposureNotification.ios.cs

```diff
@@ -60,6 +63,50 @@ static async Task<bool> PlatformIsEnabled()
 			return m.ExposureNotificationEnabled;
 		}
 
+
+		static Task PlatformScheduleFetch()
+		{
+			// This is a special ID suffix which iOS treats a certain way
+			// we can basically request infinite background tasks
+			// and iOS will throttle it sensibly for us.
+			var id = AppInfo.PackageName + ".exposure-notification";
+
+			void scheduleBgTask()
+			{
+				var newBgTask = new BGProcessingTaskRequest(id);
+				newBgTask.RequiresNetworkConnectivity = true;
+				BGTaskScheduler.Shared.Submit(newBgTask, out _);
+			}
+
+			BGTaskScheduler.Shared.Register(id, null, async t =>
+			{
+				if (await PlatformIsEnabled())
+				{
+					var cancelSrc = new CancellationTokenSource();
+					t.ExpirationHandler = cancelSrc.Cancel;
+
+					Exception ex = null;
+					try
+					{
+						await ExposureNotification.UpdateKeysFromServer();
+						t.SetTaskCompleted(true);
+					}
+					catch (Exception e)
+					{
+						ex = e;
+					}
+
+					t.SetTaskCompleted(ex != null);
```

Comment:
Can we move this into the try catch?

Reply:
Yeah typo, removed the one inside try/catch as we always want to say we're done our work even if our work failed.